### PR TITLE
Refactor quo channel component

### LINKED ISSUE #16461: Check and refactor `Channel` component
# Problem

The Channel component might not exactly match the figma designs:
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=399-20768&mode=design&t=CiGyhEQaOD0ngJLI-4

We need to make sure this component is being rendered as expected and split the code into a `style` and `view` namespaces.

We must update this component to  use the `with-theme` function.

## Acceptance Criteria

The component is rendered as expected and the code has been refactored.

### PR DESCRIPTION
**Pull request by ajayesivan**
fixes #16461 

### Summary
- Refactored channel component to match the Figma design & quo guidelines.
- Updated channel component usages

### Testing notes
Quo2 Preview -> List items -> Channel

Light theme | Dark theme
-|-
Simulator Screenshot - iPhone 11 Pro - 2023-08-21 at 21 41 00 | Simulator Screenshot - iPhone 11 Pro - 2023-08-21 at 21 40 42

#### Platforms
- Android
- iOS

#### Areas that maybe impacted
- Community Overview Screen

Light theme | Dark theme
-|-
Simulator Screenshot - iPhone 11 Pro - 2023-08-21 at 21 44 39 | Simulator Screenshot - iPhone 11 Pro - 2023-08-21 at 21 42 00

status: ready

### CODE REVIEW

File: src/quo2/components/list_items/channel/view.cljs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+   - on-press - (function, default: nil) - Function called when the component is pressed.
+   - on-long-press - (function, default: nil) - Function called when the component is long pressed.
+   - theme - Theme value from with-theme HOC"
+  [{:keys [state notification locked? mentions-count customization-color emoji name on-press
+           on-long-press theme]
+    :or   {state :default}}]
+  [rn/pressable
+   {:style               (style/container state customization-color)
+    :accessibility-label :channel-list-item
+    :on-press            on-press
```
Comment (J-Son89) [inline src/quo2/components/list_items/channel/view.cljs]: @ajayesivan can you check out how the quo2 button is handling pressed state and the colors - it uses an overlay. This correctly handles customization color (and community color) 👍

Comment (ajayesivan) [reply src/quo2/components/list_items/channel/view.cljs]: Sure. I was going to revisit the pressed state implementation when I saw your comment here https://github.com/status-im/status-mobile/pull/17074#discussion_r1301347478. I will also check the color.

Comment (ajayesivan) [reply src/quo2/components/list_items/channel/view.cljs]: @J-Son89 Done

Comment (J-Son89) [reply src/quo2/components/list_items/channel/view.cljs]: nice, curious how did it work considering the values here are opacity 5% etc? all looks good? 🤔

File: src/quo2/components/list_items/channel/style.cljs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+(defn container
+  [pressed? customization-colors]
+  {:height             48
+   :border-radius      12
+   :padding-horizontal 12
+   :padding-vertical   8
+   :align-items        :center
+   :overflow           :hidden
+   :background-color   (if pressed?
+                         (colors/alpha customization-colors 0.05)
```
Comment (J-Son89) [inline src/quo2/components/list_items/channel/style.cljs]: do we still need this? 🤔 
also `colors/custom-color` has opacity param in  in it

Comment (ajayesivan) [reply src/quo2/components/list_items/channel/style.cljs]: I was looking into it. Without opacity, it looks completely different. I will compare the resulting colors with Figma and let you know.

Comment (J-Son89) [reply src/quo2/components/list_items/channel/style.cljs]: thanks - I wonder does the overlay have to take the opacity into consideration? 🤔

Comment (J-Son89) [reply src/quo2/components/list_items/channel/style.cljs]: apologies, this is right afaik! :) - although

I think it should be 
```
colors/theme-colors 
 (colors/custom-color customization-color 50 5)
  (colors/custom-color customization-color 60 5)
  theme)
  ```
  
  the colors/custom-color also has some internal handling to figure out if it's a community color or a custom preset - i.e it checks  if it's a hex string or a keyword

Comment (ajayesivan) [reply src/quo2/components/list_items/channel/style.cljs]: Yes, that's correct. And from my understanding, we don't need to use `customization-colors/overlay` in this case.

File: src/quo2/foundations/colors.cljs [hunk 3]
```diff
@@ -256,9 +256,9 @@
             color-keyword (keyword color)
             base-color    (get-in colors-map
                                   [color-keyword suffix])]
-        (if hex?
-          color
-          (if opacity (alpha base-color (/ opacity 100)) base-color)))))))
+        (if opacity
+          (alpha (if hex? color base-color) (/ opacity 100))
+          (if hex? color base-color)))))))
```
Comment (ajayesivan) [inline src/quo2/foundations/colors.cljs]: @J-Son89 I have made a small change here to handle opacity for hexa values. Please have a look.

Comment (J-Son89) [reply src/quo2/foundations/colors.cljs]: yeah looks good, perhaps we can use `cond` to flatten the structure?

Comment (ajayesivan) [reply src/quo2/foundations/colors.cljs]: Not sure how `cond` will improve this!

Comment (J-Son89) [reply src/quo2/foundations/colors.cljs]: imo it removes the nested if's and becomes more obvious of what is being set and when. e.g something like:
```
(cond
  (and opacity hex) (alpha color (/ opacity 100))
  opacity (alpha base-color (/ opacity 100))
  hex? color
  :else base-color)
```

Comment (ajayesivan) [reply src/quo2/foundations/colors.cljs]: Done

**Review by flexsurfer — APPROVED** [review]
nice

**Comment by ibrkhalil** [comment]
I had the PR open in my browser to review and pressed update by accident, Feel free to undo it @ajayesivan :)

File: src/quo2/components/list_items/channel/style.cljs [hunk 4]
```diff
@@ -0,0 +1,42 @@
+(ns quo2.components.list-items.channel.style
+  (:require [quo2.foundations.colors :as colors]))
+
+(def ^:private label-colors
+  {:notification {:light colors/neutral-100
```
Comment (vkjr) [inline src/quo2/components/list_items/channel/style.cljs]: `colors/theme-colors` can be used here to avoid mentioning `:dark`, `:light` directly

**Review by vkjr — APPROVED** [review]

**Review by J-Son89 — APPROVED** [review]

**Comment by VolodLytvynenko** [comment]
Hi @ajayesivan. Thank you for PR. Here a list of found issues: 
## ISSUE 1: Missing space for community channels between '#' and channel name

### Steps to reproduce:
1. Open any community
2. Check channel

### Actual result:
The space is missing between '#' and channel name
image

### Expected result:
The channels are with spaces between '#' and channel name
image
https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=411-18564&mode=design&t=fRO1mgNKgA83V5be-0

**Comment by VolodLytvynenko** [comment]
## ISSUE 2:  Missing locks badges on community channels
### Steps:
1. Open any community

### Actual result:
 Missing of locks for community channels
### Expected result:
The locks are shown
image

**Comment by ajayesivan** [comment]
@VolodLytvynenko Fixed both the issues. Could you please test again? Thanks ❤️ .

**Comment by VolodLytvynenko** [comment]
Hi @ajayesivan, thank you for the PR. I don't see any issues from my side. Please don't merge this PR until the e2e tests are rerun. I'll let you know when the PR can be merged. Thank you

@Francesca-G Could you please take a look at this PR?
The known issue:
- The lock badges are always shown for all kinds of channels

**Review by Francesca-G — COMMENTED** [review]
Here's the [review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=5212%3A43346&mode=design&t=1jDQwMUuBqfSdLZO-1) :)

**Comment by yevh-berdnyk** [comment]
hi @ajayesivan, may I ask you to add the accessibility id for a channels' unread messages indicator the same as  it was before and as it is for communities element:
image

For a channel  element it's just 'icon' now:
image
Thanks!

**Comment by ajayesivan** [comment]
@yevh-berdnyk I have added the accessibility label back. Feel free to let me know if you need anything else from me.

**Comment by ajayesivan** [comment]
@Francesca-G I've addressed the design review issues. Kindly take another look, and let me know if you come across any other issues.

**Comment by VolodLytvynenko** [comment]
@ajayesivan No issues from my side. e2e run related to this PR is done in https://github.com/status-im/status-mobile/pull/17172. Thank you for your work

**Comment by yevh-berdnyk** [comment]
> @ajayesivan No issues from my side. e2e run related to this PR is done in #17172. Thank you for your work

I've just restarted  the  tests with the added accessibility label, the  results will  be here

**Review by Francesca-G — APPROVED** [review]

**Comment by ajayesivan** [comment]
Hi @yevh-berdnyk, can I merge this PR?

**Comment by yevh-berdnyk** [comment]
> Hi @yevh-berdnyk, can I merge this PR?

Yes from my side :)

**Pull request merged**